### ISSUE TITLE
Ray does not preserve requires_grad attribute

### ISSUE THREAD

**Issue by timtody** [ray-project/ray#6405]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Ray installed from (source or binary)**: pip install ray
- **Ray version**: 0.7.6
- **Python version**: Python 3.6.9
- **Exact command to reproduce**:

```
import torch
import ray

@ray.remote
def foo(tensor):
    return tensor

t = torch.tensor(1., requires_grad=True)

ray.init()
result = foo.remote(t)
print(ray.get(result).requires_grad) # returns False
```

### Describe the problem
When using tensors with the @ray.remote decorator, the requires_grad attribute is not preserved.

**Comment by pcmoritz** [comment]
Thanks for the report @timtody! This is already fixed on the latest master and will be part of the upcoming 0.7.7 release:

```
In [1]: import torch                                                                                                                                                                                                                                          

In [2]: import ray                                                                                                                                                                                                                                            

In [3]: @ray.remote 
   ...: def foo(tensor): 
   ...:     return tensor 
   ...:  
   ...: t = torch.tensor(1., requires_grad=True)                                                                                                                                                                                                              

In [4]: ray.init() 
   ...: result = foo.remote(t) 
   ...: print(ray.get(result).requires_grad)
True
```

**Comment by pcmoritz** [comment]
If you want to use a version that has the fix included, you can use https://ray.readthedocs.io/en/latest/installation.html#latest-snapshots-nightlies :)

**Comment by timtody** [comment]
Hey, thanks for coming back to me! I installed Version 0.8.0.dev7 from your list of available wheels at https://ray.readthedocs.io/en/latest/installation.html#latest-snapshots-nightlies and the requires_grad attribute is now correctly preserved. Gradients themselves are still not preserved though:

```
import torch
import ray

@ray.remote
def foo(tensor):
    tensor.backward()
    print(tensor.grad) # (pid=7481) tensor(1.)
    return tensor

t = torch.tensor(1., requires_grad=True)

ray.init()
result = foo.remote(t)
print(ray.get(result).grad) # None
```

Which seems to me that the underlying problem is probably of bigger scope...

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
